pydantic/fields.py

```diff
@@ -364,6 +364,9 @@ def _type_analysis(self) -> None:  # noqa: C901 (ignore complexity)
         origin = getattr(self.type_, '__origin__', None)
         if origin is None:
             # field is not "typing" object eg. Union, Dict, List etc.
+            # allow None for virtual superclasses of NoneType, e.g. Hashable
+            if isinstance(self.type_, type) and isinstance(None, self.type_):
```

Comment:
Are there any other virtual superclasses of `NoneType`? Is there a reference of this behavior anywhere?

Reply:
I'm not aware of other virtual subclasses, however the one non-virtual superclass of `NoneType` - `object` - is also handled by this change. 

The behavior is provided by [ABCMeta.register](https://docs.python.org/3/library/abc.html#abc.ABCMeta.register) or the [ABCMeta.__subclasshook__](https://docs.python.org/3/library/abc.html#abc.ABCMeta.__subclasshook__) classmethod. Toy example:
```py

In [1]: from abc import ABCMeta                                                                                                                                          

In [2]: class Singleton(metaclass=ABCMeta): 
   ...:     pass 

In [3]: for obj in None, Ellipsis, NotImplemented, True, False: 
   ...:     Singleton.register(type(obj)) 

In [4]: isinstance(None, Singleton)                                                                                                                                      
Out[4]: True

In [5]: isinstance(True, Singleton)                                                                                                                                      
Out[5]: True

In [6]: isinstance(1, Singleton)                                                                                                                                         
Out[6]: False
```